packages/components/calendar/src/calendar-cell.tsx

```diff
@@ -49,9 +49,9 @@ export function CalendarCell(originalProps: CalendarCellProps) {
     !isDisabled && !isInvalid && state.isCellUnavailable(props.date.subtract({days: 1}));
   const highlightedRange = "highlightedRange" in state && state.highlightedRange;
   const isSelectionStart =
-    isSelected && highlightedRange && isSameDay(props.date, highlightedRange.start);
+    isSelected && highlightedRange ? isSameDay(props.date, highlightedRange.start) : false;
```

Comment:
Due to the type change, `highlightedRange` can be `null`.
https://github.com/adobe/react-spectrum/pull/6076